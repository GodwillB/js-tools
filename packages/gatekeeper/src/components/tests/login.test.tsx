import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import OauthLogin, { ProviderLinks } from '../login';
import * as fixtures from './fixtures';

describe('gatekeeper/OauthLogin', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const props = {
      providers: fixtures.providers
    };
    shallow(<OauthLogin {...props} />);
  });

  it('renders correctly', () => {
    const props = {
      providers: fixtures.providers
    };
    const wrapper = mount(<OauthLogin {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders correctly with custom ProviderLinksComponent', () => {
    const CustomProviderLinks = () => {
      return (
        <div className="danger">
          <p className="closed">oops!</p>
        </div>
      );
    };

    const props = {
      ProviderLinksComponent: CustomProviderLinks,
      providers: fixtures.providers
    };
    const wrapper = mount(<OauthLogin {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
