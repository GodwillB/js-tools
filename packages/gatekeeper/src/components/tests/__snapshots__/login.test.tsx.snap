// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatekeeper/OauthLogin renders correct links for Authorization code-authorization grant flow: Authorization code authorization grant 1`] = `
<a
  className="gatekeeper-btn"
  href="https://stage-api.ona.io/o/authorize/?client_id=hunter2&redirect_uri=https%3A%2F%2Fexample.com%2Foauth%2Fcallback%2Fonadata%2F&scope=read%20write&response_type=code&state=abc"
>
  onadata
</a>
`;

exports[`gatekeeper/OauthLogin renders correct links for implicit authorization grant flow: implicit authorization grant 1`] = `
<a
  className="gatekeeper-btn"
  href="https://stage-api.ona.io/o/authorize/?client_id=hunter2&redirect_uri=https%3A%2F%2Fexample.com%2Foauth%2Fcallback%2Fonadata%2F&scope=read%20write&response_type=token&state=abc"
>
  onadata
</a>
`;

exports[`gatekeeper/OauthLogin renders correctly with custom ProviderLinksComponent: custom provider link 1`] = `
<div
  className="danger"
>
  <p
    className="closed"
  >
    oops!
  </p>
</div>
`;

exports[`gatekeeper/OauthLogin renders correctly with no ProviderLinksComponent 1`] = `
<div
  className="gatekeeper-login"
>
  <p
    className="gatekeeper-p"
  >
    Please log in with one of the following providers
  </p>
  <p
    className="gatekeeper-p item"
    key="onadata"
  >
    <a
      className="gatekeeper-btn"
      href="https://stage-api.ona.io/o/authorize/?client_id=hunter2&redirect_uri=https%3A%2F%2Fexample.com%2Foauth%2Fcallback%2Fonadata%2F&scope=read%20write&response_type=token&state=abc"
    >
      onadata
    </a>
  </p>
</div>
`;
